src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -434,14 +434,24 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state, GCHand
                 // Set maximum automatic redirection option
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_MAXREDIRS, _maxAutomaticRedirections);
             }
-            if (state.RequestMessage.Content != null)
+
+            if (state.RequestMessage.Method == HttpMethod.Put)
```

Comment:
Curl do support a Put/Post with no content , so the change here is to follow that. checking content if null or not, is not correct here.